### PR TITLE
chore(api): remove ValidateDeleteRequestsListener

### PR DESCRIPTION
**Pull request by usu**
Related to https://github.com/ecamp/ecamp3/issues/5095

In the meantime, api platform supports validation for delete operation. Hence we can remove our own listener and enable validation via api-platform configuration.

**Commit 1:** https://github.com/ecamp/ecamp3/commit/ea8c46e317bb3e50f7729ab0f52c13f70e424e54
Removes our own listener
Failing tests proof that we have proper test for each endpoint which requires delete validation:
https://github.com/ecamp/ecamp3/actions/runs/9329296485/job/25681554883?pr=5275

**Commit 2:** https://github.com/ecamp/ecamp3/pull/5275/commits/009ae7bb9ed88a82ddeea1761430a8f65f8fb8ed
Enables delete validation for relevant endpoints

### CODE REVIEW

**Comment by usu** [comment]
The changes in OpenApi doc is probably related to https://github.com/api-platform/core/issues/6360

File: api/tests/Api/SnapshotTests/__snapshots__/ResponseSnapshotTest__testOpenApiSpecMatchesSnapshot__1.yml [hunk 1]
```diff
@@ -5330,42 +5330,26 @@ components:
         id:
           description: 'An internal, unique, randomly generated identifier of this entity.'
           example: 1a2b3c4d
-          maxLength: 16
```
Comment (BacLuc) [inline api/tests/Api/SnapshotTests/__snapshots__/ResponseSnapshotTest__testOpenApiSpecMatchesSnapshot__1.yml:5333]: ?
ok, what happened here?

File: api/tests/Api/SnapshotTests/__snapshots__/ResponseSnapshotTest__testOpenApiSpecMatchesSnapshot__1.yml [hunk 2]
```diff
@@ -5330,42 +5330,26 @@ components:
...
-          maxLength: 128
-          minLength: 1
           type:
             - 'null'
             - string
         role:
           description: |-
             The role that this person has in the camp. Depending on the role, the collaborator might have
             different access rights. There must always be at least one manager in a camp.
-          enum:
```
Comment (BacLuc) [inline api/tests/Api/SnapshotTests/__snapshots__/ResponseSnapshotTest__testOpenApiSpecMatchesSnapshot__1.yml:5353]: and here?

File: api/tests/Api/SnapshotTests/__snapshots__/ResponseSnapshotTest__testOpenApiSpecMatchesSnapshot__1.yml (hunk 1 above)
Comment (usu) [reply api/tests/Api/SnapshotTests/__snapshots__/ResponseSnapshotTest__testOpenApiSpecMatchesSnapshot__1.yml:5333]: See comment above:

> The changes in OpenApi doc is probably related to [api-platform/core#6360](https://github.com/api-platform/core/issues/6360)

**Review by BacLuc — APPROVED** [review]

File: api/src/Entity/MaterialList.php [hunk 3]
```diff
@@ -34,7 +34,9 @@
             security: 'is_granted("CAMP_MEMBER", object) or is_granted("CAMP_MANAGER", object)'
         ),
         new Delete(
-            security: 'is_granted("CAMP_MEMBER", object) or is_granted("CAMP_MANAGER", object)'
+            security: 'is_granted("CAMP_MEMBER", object) or is_granted("CAMP_MANAGER", object)',
+            validate: true,
+            validationContext: ['groups' => ['delete']]
```
Comment (manuelmeister) [inline api/src/Entity/MaterialList.php:39]: Where is this group set?

Comment (usu) [reply api/src/Entity/MaterialList.php:39]: We use it a bit further below in Line 62 in the attribute for the validation constraint:
https://github.com/ecamp/ecamp3/pull/5275/files#diff-4e6f5c93038c615b8c2dc034a317af9338b08371dda797697fff316b0b9a819fR62

See also the example in the api-platform docs:
https://api-platform.com/docs/core/validation/#using-validation-groups-on-operations

Or specifically on delete operation:
https://api-platform.com/docs/guides/delete-operation-with-validation/

**Review by pmattmann — APPROVED** [review]

**Pull request merged**